tests/manage/cluster/nodes/test_nodes_restart.py

```diff
@@ -0,0 +1,200 @@
+import logging
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.utility.utils import ceph_health_check
+from ocs_ci.ocs import ocp, constants
+from ocs_ci.framework.testlib import tier4, ignore_leftovers, ManageTest
+from ocs_ci.utility import aws
+from ocs_ci.ocs.cluster import CephCluster
+from tests import helpers
+
+
+logger = logging.getLogger(__name__)
+
+AWS = aws.AWS()
+
+
+@pytest.fixture()
+def instances(request):
+    """
+    Get cluster instances
+
+    Returns:
+        dict: The ID keys and the name values of the instances
+    """
+    def finalizer():
+        """
+        Make sure all instances are running
+        """
+        AWS.start_ec2_instances(instances=instances, wait=True)
+
+    request.addfinalizer(finalizer)
+
+    nodes = ocp.get_all_nodes()
+    instances = aws.get_instances_ids_and_names(nodes)
+    return instances
+
+
+@pytest.fixture()
+def resources(request):
+    """
+    Delete the resources created during the test
+
+    Returns:
+        tuple: empty lists of resources
+
+    """
+    projects = list()
+    secrets = list()
+    pools = list()
+    storageclasses = list()
+    pvcs = list()
+    pods = list()
+
+    def finalizer():
+        """
+        Delete the resources created during the test
+        """
+        if pods:
+            for pod in pods:
+                pod.delete()
+                pod.ocp.wait_for_delete(pod.name)
+        if pvcs:
+            for pvc in pvcs:
+                pvc.delete()
+                pvc.ocp.wait_for_delete(pvc.name)
+        if storageclasses:
+            for sc in storageclasses:
+                sc.delete()
+                sc.ocp.wait_for_delete(sc.name)
```

Comment:
for resource in zip(pods, pvcs, storageclasses, secrets):
  resources.delete()
  resource.ocp.wait_for_delete(resource.name)

Reply:
we can't do what you've suggested (even though it's very cool) because pools[1] is an existing FS pool that we just get its name and don't want to delete

Reply:
I'm not sure I understood, but please do note that the pools aren't included in this - I only minified pods, pvcs, storageclasses and secrets :)

I noticed pools and projects have different deletion procedures and I suggest to leave them as they are (and unify all the ones who use the same methodology)

Reply:
now I see that :)

Regarding zip usage - it iterates over the tuple and doesn't get into each item in each of the tuple lists
Instead, I'll iterate over each of the tuple lists